packages/core/src/service/helpers/shipping-calculator/shipping-calculator.ts

```diff
@@ -6,6 +6,15 @@ import { Order } from '../../../entity/order/order.entity';
 import { ShippingMethod } from '../../../entity/shipping-method/shipping-method.entity';
 import { ShippingMethodService } from '../../services/shipping-method.service';
 
+type CheckEligibilityByShippingMethodResponse =
```

Comment:
I would advise to just define a single type `CheckEligibilityByShippingMethodResponse ` without the `undefined` union, and then in define like `checkEligibilityByShippingMethod(): Promise<CheckEligibilityByShippingMethodResponse | undefined>`. This requires one less type definition and IMO is simpler to understand.

Reply:
Nice! Good point. I will update